pkg/kotsadmconfig/validation/validator.go

```diff
@@ -0,0 +1,65 @@
+package validation
+
+import (
+	kotsv1beta1 "github.com/replicatedhq/kots/kotskinds/apis/kots/v1beta1"
+	configtypes "github.com/replicatedhq/kots/pkg/kotsadmconfig/types"
+)
+
+var (
+	validatableItemTypesMap = map[string]bool{
+		configtypes.EmptyItemType:    true,
+		configtypes.TextItemType:     true,
+		configtypes.PasswordItemType: true,
+		configtypes.TextAreaItemType: true,
+		configtypes.FileItemType:     true,
+	}
+)
+
+func isValidatableConfigItem(item kotsv1beta1.ConfigItem) bool {
+	if item.Validation == nil {
+		return false
+	}
+
+	if item.Hidden {
+		return false
+	}
+
+	if item.When == "false" {
+		return false
+	}
+
+	if item.Repeatable {
+		return false
+	}
+
+	if !validatableItemTypesMap[item.Type] {
+		return false
+	}
+
+	return true
+}
+
+func validate(value string, itemValidation kotsv1beta1.ConfigItemValidation) ([]configtypes.ValidationError, error) {
+	var validationErrs []configtypes.ValidationError
+	validators := buildValidators(itemValidation)
+	for _, v := range validators {
+		validationErr, err := v.Validate(value)
+		if err != nil {
+			return nil, err
```

Comment:
error wrap

Reply:
un-resolving this conversation since it seems like this wasn't addressed.  let me know if there was a particular reason to leave it as-is